cmd/monitorCodeGen/main.go

```diff
@@ -0,0 +1,195 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"text/template"
+)
+
+var tmpl = `// Code generated by monitor-code-gen. DO NOT EDIT.
+
+package {{.goPackage}}
+
+import "github.com/signalfx/signalfx-agent/internal/monitors"
+
+{{with .groupMetricsMap}}
+const (
+{{- range . }}
+	{{(printf "group.%s" .Name) | formatVariable}} = "{{.Name}}"
+{{- end}}
+)
+{{end}}
+
+var groupSet = map[string]bool {
+{{- range .groupMetricsMap}}
+	{{(printf "group.%s" .Name) | formatVariable}}: true,
```

Comment:
Same thing here; if the variable name for the map is already `groupSet` why does there need to be a `group.` prefix for  all the key names?  Also I don't understand how this generates valid map literals since the keys don't seem to be in double quotes in the output Go src.

Reply:
Here's what it looks like:

```go
const (
	groupContainers = "containers"
	groupDaemonSets = "daemon-sets"
	groupDeployments = "deployments"
	groupNamespaces = "namespaces"
	groupNodes = "nodes"
	groupPods = "pods"
	groupReplicaSets = "replica-sets"
	groupReplicationControllers = "replication-controllers"
	groupResourceQuotas = "resource-quotas"
)


var groupSet = map[string]bool {
	groupContainers: true,
	groupDaemonSets: true,
	groupDeployments: true,
	groupNamespaces: true,
	groupNodes: true,
	groupPods: true,
	groupReplicaSets: true,
	groupReplicationControllers: true,
	groupResourceQuotas: true,
}
```

Reply:
oh ok, I see, it just references the constants.